### PR TITLE
[Feat] YCSB stat

### PR DESCRIPTION
**Pull request by IHEII**
## Summary

- Show details when in progress of YCSB

## Solution Description

- Show details when in progress of YCSB

### CODE REVIEW

File: ycsb-rs/src/main.rs [hunk 1]
```diff
@@ -64,9 +73,11 @@ async fn run_ob(
     db: Arc<OBKVClient>,
     rng: Arc<Mutex<SmallRng>>,
     operation_count: usize,
+    counter: Arc<AtomicUsize>,
 ) {
     for _ in 0..operation_count {
         wl.ob_transaction(rng.clone(), db.clone()).await;
+        counter.fetch_add(1, Ordering::SeqCst);
```
Comment (Rachelint) [inline ycsb-rs/src/main.rs:80]: If just use it as a counter, order can be set to `Ordering::Relaxed`.

File: ycsb-rs/src/main.rs [hunk 2]
```diff
@@ -106,22 +127,66 @@ fn main() -> Result<()> {
...
                 let db = db::create_ob(&database, config.clone()).unwrap();
+                // count the ops per client
+                let counter = Arc::new(AtomicUsize::new(0));
+                db_counters.push(counter.clone());
                 for _ in 0..props.obkv_client_reuse {
                     let db = db.clone();
                     let wl = wl.clone();
                     let cmd = cmd.clone();
                     let runtime = runtimes.default_runtime.clone();
+                    let counter_clone = Arc::clone(&counter);
```
Comment (Rachelint) [inline ycsb-rs/src/main.rs:138]: `Arc::clone` and `.clone` are all ok.
Seem `.clone` is used above, maybe follow the same style here?

File: ycsb-rs/src/properties.rs [hunk 3]
```diff
@@ -4,6 +4,14 @@ fn zero_u64() -> u64 {
     0
 }
 
+fn false_bool() -> bool {
+    false
+}
```
Comment (Rachelint) [inline ycsb-rs/src/properties.rs:9]: show_prometheus_default ?

File: ycsb-rs/src/main.rs [hunk 4]
```diff
@@ -106,22 +127,66 @@ fn main() -> Result<()> {
...
+            // show progress
+            let stat_duration_sec = props.show_progress_duration;
+            tasks.push(runtimes.default_runtime.spawn(async move {
+                let mut interval = tokioTime::interval(Duration::from_secs(stat_duration_sec));
+                let mut prev_count = 0;
+                loop {
+                    interval.tick().await;
+                    let completed_operations: usize = db_counters
+                        .iter()
+                        .map(|arc| arc.load(Ordering::SeqCst))
```
Comment (Rachelint) [inline ycsb-rs/src/main.rs:164]: counter?

File: ycsb-rs/src/main.rs [hunk 5]
```diff
@@ -106,22 +127,66 @@ fn main() -> Result<()> {
...
                 for _ in 0..props.obkv_client_reuse {
                     let db = db.clone();
                     let wl = wl.clone();
                     let cmd = cmd.clone();
                     let runtime = runtimes.default_runtime.clone();
+                    let counter_clone = counter.clone();
                     tasks.push(runtime.spawn(async move {
                         let rng = Arc::new(Mutex::new(SmallRng::from_entropy()));
                         db.init().unwrap();
                         match &cmd[..] {
```
Comment (Rachelint) [inline ycsb-rs/src/main.rs:142]: cmd.as_str()

File: ycsb-rs/src/main.rs [hunk 6]
```diff
@@ -106,22 +127,66 @@ fn main() -> Result<()> {
             for _client_idx in 0..actual_client_count {
```
Comment (Rachelint) [inline ycsb-rs/src/main.rs:127]: _client_idx --> _

File: ycsb-rs/src/main.rs (hunk 1 above)
Comment (IHEII) [reply ycsb-rs/src/main.rs]: Fixed

File: ycsb-rs/src/main.rs (hunk 2 above)
Comment (IHEII) [reply ycsb-rs/src/main.rs]: Nice suggestion, fixed

File: ycsb-rs/src/properties.rs (hunk 3 above)
Comment (IHEII) [reply ycsb-rs/src/properties.rs]: Fixed

File: ycsb-rs/src/main.rs [hunk 7]
```diff
@@ -12,6 +14,7 @@ use obkv::dump_metrics;
 use properties::Properties;
 use rand::{rngs::SmallRng, SeedableRng};
 use structopt::StructOpt;
+use tokio::time as tokioTime;
```
Comment (Rachelint) [inline ycsb-rs/src/main.rs:17]: TokioTime

File: ycsb-rs/src/main.rs (hunk 4 above)
Comment (IHEII) [reply ycsb-rs/src/main.rs]: Fixed

**Pull request merged**